### ISSUE TITLE
Temps printed out in the wrong order?

### ISSUE THREAD

**Issue by bsodmike** [murraytodd/embassy-rp-blinky#1]
Hi!

Just been going through your excellent work and spotted this https://github.com/murraytodd/embassy-rp-blinky/blob/main/src/main.rs#L256-L261 - looks like the order needs to be switched around?

```rs
        info!(
            "Temp readings:  MCP9808: {}°F, TMP36: {}°F, OnChip: {}°F",
            c_to_f(mcp9808_reading_c),
            chip_f(chip_voltage_24bit),
            tmp36_f(tmp36_voltage_24bit)
        );
```

Thanks,
Mike.

**Comment by murraytodd** [comment]
Sharp eyes! Thanks, and it’s great to know someone’s looking through my code.

Murray Williams

**Comment by bsodmike** [comment]
Haha! I swear most days I'm seeing squares but your work on rp is fantastic, specially getting networking going. 

I need to pick up a Pico W to test this out though, but for now I've used your example to test out an embedded Hal driver I've been cleaning up.

Best, Mike.

**Comment by murraytodd** [comment]
@bsodmike I've been working on a lot of refactoring for both the networking and sensing code. (I figured it would both be good to _try_ doing this in Rust as a learning experience and then write about some of my findings with the language.) I'm going to go ahead and commit some of this interim work to the master branch. If you feel like looking anything over and if you have any advice (this is clumsy, that's involving too much boilerplate for addressing errors, etc.) you're more than welcome to comment. I'm trying to learn the language in a little bit of a vacuum.

**Comment by bsodmike** [comment]
Will do Murray, just ping me here once you merge and I'm happy to go through it.

**Comment by murraytodd** [comment]
Okay, I've got more work to do down the road before I'm ready to write my next blog, but as I said--I'd love to get an early opinion of things. I haven't gotten to moving much networking functionality out of main.rs and into networking.rs, but I did everything I could to both move sensor-specific functionality into sensors.rs and tried to the do the sorts of abstractions (creating the concepts of a TempReading, etc.) that I would normally do as a Scala programmer.

The other thing I was trying to do (for my next blog) was to tackle weaning myself away from lazy .unwrap() calls and working with Result values more elegantly. Not much, but you can see some stuff in main.rs from lines 200-235 where I show using .map() to do some clean data manipulation to the happy-path data while keeping the Result structures intact.

Any comments or advice is welcome! (Also, I'm not sure if I'm going the right direction with error.rs.)

**Comment by murraytodd** [comment]
Fixed in commit c9fd925

**Comment by bsodmike** [comment]
Hi Murray,

Can you look at https://github.com/bsodmike/rv8803-rs/pull/20 - I'm cleaning this up as a I go but the way I've organised my public API is something I've focussed on.

Also do run the build script and look at the docs.  FYI, there are some generic stuff that might do your head in, so don't focus on the traits etc (that's hard for those uninitiated) but more on the organisation, and separating things into modules with pub/private etc.
